### PR TITLE
Redundant Read/Write Slices

### PR DESCRIPTION
**Pull request by alexnick83**
This PR introduces the RedundantReadSlice and RedundantWriteSlice transformations. We define slices as DaCe Views of DaCe Arrays following the same semantics as NumPy Views generated by slicing NumPy Arrays with basic indexing. The new transformations are essentially simplified versions of RedundantArray and RedundantSecondArray with much stricter constraints. Furthermore, `SDFG.apply_strict_transformations` makes a first pass over the new transformations (only) in order to simplify the graph as much as possible before continuing with the other strict transformations.

### CODE REVIEW

File: dace/transformation/dataflow/redundant_array.py [hunk 1]
```diff
@@ -1161,3 +1176,339 @@ def apply(self, sdfg: SDFG):
             sdfg.remove_data(in_array.data)
         except ValueError:  # Already in use (e.g., with Views)
             pass
+
+
+def _is_slice(adesc: data.Array, vdesc: data.View) -> bool:
```
Comment (tbennun) [inline dace/transformation/dataflow/redundant_array.py:1181]: Nice, I like this function, we could make it a general utility

File: dace/transformation/dataflow/redundant_array.py [hunk 2]
```diff
@@ -1161,3 +1176,339 @@ def apply(self, sdfg: SDFG):
...
+
+        # The match must be View(Array) -> Array , i.e.,
+        # the View must point to the Array. Find the true in_desc.
+        true_in_array = sdutil.get_last_view_node(graph, in_array)
+        if not true_in_array:
+            return False
+        if true_in_array is not out_array:
+            return False
+        
+        # If the View receives data from a reduction, fail.
```
Comment (tbennun) [inline dace/transformation/dataflow/redundant_array.py:1410]: :(

**Review by tbennun — APPROVED** [review]
LGTM!

File: dace/transformation/dataflow/redundant_array.py (hunk 2 above)
Comment (alexnick83) [reply dace/transformation/dataflow/redundant_array.py:1410]: The reduce node must be updated at some point I guess. The issue still exists.

**Pull request merged**